pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -98,11 +145,64 @@ type DockercfgController struct {
 
 	serviceAccountCache      MutationCache
 	serviceAccountController *framework.Controller
+	secretsIndex             cache.Indexer
+	secretController         *framework.Controller
 
 	queue workqueue.RateLimitingInterface
 
 	// syncHandler does the work. It's factored out for unit testing
 	syncHandler func(serviceKey string) error
+
+	// record the token secret we created to pendingTokens and prevent creating a new tokens
+	// while the cache is populated
+	pendingTokens *pendingTokenStore
+}
+
+// handleTokenSecretUpdate checks if the service account token secret is populated with
+// token data and triggers re-sync of service account when the data are observed.
+func (e *DockercfgController) handleTokenSecretUpdate(_, newObj interface{}) {
+	secret := newObj.(*api.Secret)
```

Comment:
need to check the createdBy annotation early... this shouldn't do anything if it isn't one of our tokens